src/common/executor/ExecutorOutputQueue.js

```diff
@@ -0,0 +1,120 @@
+/*globals define*/
+/*jshint node:true*/
+/**
+ * @author pmeijer / https://github.com/pmeijer
+ */
+
+define([], function () {
+    'use strict';
+
+    function ExecutorOutputQueue(worker, jobInfo, interval, segmentSize) {
+        var self = this;
+        this.logger = worker.logger; //TODO: Consider using gmeLogger for ExecutorWorker
+        this.jobInfo = jobInfo;
+
+        this.currOutput = {
+            lines: [],
+            callback: null,
+            finishFn: null
+        };
+        this.outputQueue = [];
+
+        this.timeoutId = null;
+
+        /**
+         * Adds a string to the current output (which will be queued based on interval and/or segmentSize).
+         * @param {string} outputStr
+         */
+        this.addOutput = function (outputStr) {
+            var lines = outputStr.split(/\r\n|\r|\n/);
+
+            self.currOutput.lines = self.currOutput.lines.concat(lines);
+            self.logger.debug('length', self.currOutput.lines.length);
+            if (segmentSize > -1 && self.currOutput.lines.length >= segmentSize) {
```

Comment:
shouldn't it be <=segmentSize? 
(and if it is, shouldn't we avoid 0 segmentSize?)